help_function/src/onedpl_test_vector.cpp

```diff
@@ -209,6 +256,50 @@ int main() {
 
         failed_tests += test_passed(num_failing, test_name);
     }
+    num_failing = 0;
+    test_name = "custom_allocator default construction";
+    { // test with custom allocator which constructs default constructor of 6
+        dpct::device_vector<int64_t, my_allocator_with_custom_construct<int64_t>> default_construct(5);
+        default_construct[4] += 2;
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, default_construct[0], 6);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, default_construct[1], 6);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, default_construct[2], 6);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, default_construct[3], 6);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, default_construct[4], 8);
+        failed_tests += test_passed(num_failing, test_name);
+    }
+    num_failing = 0;
+    test_name = "custom_allocator construction from input";
+    { // test with custom allocator which whos default constructor adds of 3 when constructing from a value or iterator
+        dpct::device_vector<int64_t, my_allocator_with_custom_construct<int64_t>> construct_from_value(5, 2);
+        construct_from_value[4] += 2;
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_value[0], 5);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_value[1], 5);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_value[2], 5);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_value[3], 5);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_value[4], 7);
+
+        std::vector<int64_t> src(8);
+        src[0] = -1; src[1] = 2; src[2] = -3; src[3] = 4; src[4] = -5; src[5] = 6; src[6] = -7; src[7] = 8;
+
+        dpct::device_vector<int64_t, my_allocator_with_custom_construct<int64_t>> construct_from_iter(src.begin()+2, src.begin()+7);
+        construct_from_iter[4] += 2;
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_iter[0], 0);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_iter[1], 7);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_iter[2], -2);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_iter[3], 9);
+        num_failing += ASSERT_ARRAY_EQUAL(test_name, construct_from_iter[4], -2);
+        failed_tests += test_passed(num_failing, test_name);   
+    }
+
+    num_failing = 0;
+    test_name = "inserting host iterators";
```

Comment:
Is there more here to be added to this test?

Reply:
Yes... not sure how it wound up like this in the PR, but fixed.